File: Assets/Bundles/Shared/Resources/TS/Util/LayerUtil.lua
-- Compiled with unity-ts v2.1.0-75
local LayerUtil
do
	LayerUtil = setmetatable({}, {
		__tostring = function()
			return "LayerUtil"
		end,
	})
	LayerUtil.__index = LayerUtil
	function LayerUtil.new(...)
		local self = setmetatable({}, LayerUtil)
		return self:constructor(...) or self
	end
	function LayerUtil:constructor()
	end
	function LayerUtil:GetLayerMask(layers)
		local mask = 1
		local _layers = layers
		local _arg0 = function(layerValue)
			mask = bit32.bor(mask, bit32.lshift(1, layerValue))
			return mask
		end
		for _k, _v in _layers do
			_arg0(_v, _k - 1, _layers)
		end
		return mask
	end
	function LayerUtil:LayerIsInMask(layer, layerMask)
		local result = layerMask == (bit32.bor(layerMask, (bit32.lshift(1, layer))))
		return result
	end
end
return {
	LayerUtil = LayerUtil,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/AirshipPackages/@Easy/Core/Shared/Util/LayerUtil.ts
export class LayerUtil {
	public static GetLayerMask(layers: number[]): number {
		let mask = 0;

		layers.forEach((layerValue) => (mask |= 1 << layerValue));

		return mask;
	}

	public static LayerIsInMask(layer: number, layerMask: number): boolean {
		const result = layerMask === (layerMask | (1 << layer));

		return result;
	}
}
